packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -36,124 +48,131 @@ describe("VisibilityTreeEventHandler", () => {
     dispose: sinon.fake(),
   };
 
-  const testVisibilityStatus: VisibilityStatus = {
-    state: "visible",
-    isDisabled: false,
-  };
+  before(async () => {
+    await NoRenderApp.startup();
+    await TestUtils.initialize();
+  });
 
-  beforeEach(() => {
-    getVisibilityStatus.returns(testVisibilityStatus);
-    modelMock.setup((x) => x.getNode(moq.It.isAny())).returns(() => createSimpleTreeModelNode());
-    modelSourceMock.setup((x) => x.onModelChanged).returns(() => new BeUiEvent<[TreeModel, TreeModelChanges]>());
-    modelSourceMock.setup((x) => x.getModel()).returns(() => modelMock.object);
-    nodeLoaderMock.setup((x) => x.dataProvider).returns(() => dataProviderMock.object);
-    nodeLoaderMock.setup((x) => x.modelSource).returns(() => modelSourceMock.object);
+  after(async () => {
+    TestUtils.terminate();
+    await IModelApp.shutdown();
   });
 
   afterEach(() => {
-    modelSourceMock.reset();
-    modelMock.reset();
-    nodeLoaderMock.reset();
-    dataProviderMock.reset();
-    selectionHandlerMock.reset();
-    getVisibilityStatus.reset();
     changeVisibility.reset();
+    onVisibilityChange.clear();
+    getVisibilityStatus.resetHistory();
+    sinon.restore();
   });
 
+  function setupTreeModel(nodeIds: string[], item?: PresentationTreeNodeItem) {
+    const model = new MutableTreeModel();
+    model.setChildren(
+      undefined,
+      nodeIds.map((nodeId) => {
+        return { ...createSimpleTreeModelNode(nodeId), isLoading: false, item: item ?? createSimpleTreeModelNode("node-item") };
+      }),
+      0,
+    );
+    const modelSource = new TreeModelSource(model);
+
+    nodeLoaderStub = {
+      loadNode: sinon.stub<Parameters<ITreeNodeLoader["loadNode"]>, ReturnType<ITreeNodeLoader["loadNode"]>>(),
+      modelSource,
+      dataProvider: {} as any as PresentationTreeDataProvider,
+    } as any as AbstractTreeNodeLoaderWithProvider<PresentationTreeDataProvider>;
+
+    return { modelSource, nodeLoader: nodeLoaderStub as unknown as AbstractTreeNodeLoaderWithProvider<PresentationTreeDataProvider> };
+  }
+
   const createHandler = (partialProps?: Partial<VisibilityTreeEventHandlerParams>): VisibilityTreeEventHandler => {
-    if (!partialProps)
-      partialProps = {};
+    if (!partialProps) partialProps = {};
     const props: VisibilityTreeEventHandlerParams = {
       visibilityHandler: partialProps.visibilityHandler || visibilityHandler,
-      nodeLoader: partialProps.nodeLoader || nodeLoaderMock.object,
-      selectionHandler: partialProps.selectionHandler || selectionHandlerMock.object,
+      nodeLoader: partialProps.nodeLoader || (nodeLoaderStub as unknown as AbstractTreeNodeLoaderWithProvider<PresentationTreeDataProvider>),
```

Comment:
Looks like almost all tests pass visibility handler and node loader. Maybe it would be worth changing props so that `visibilityHandler` and `nodeLoader` are not optiona. In that case you would not need `nodeLoaderStub` in the root `describe`.

Reply:
I am not so sure if it is worth it, since for this to work one would need to make a separate type for this case and if someone would want to write more tests in this file he would be forced to always define these parameters beforehand which is not great.  I think we would be taking extra steps just to make it worse here.

Reply:
Changed it to only have nodeLoader as a property, since in all tests visibilityHandler and selectionHandler the same global ones are used.